modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -1050,12 +1050,17 @@ void WindowContextTop::process_configure(GdkEventConfigure* event) {
                                              geometry.final_height.value > 1)) {
         // skip artifact
         return;
+
+    }
+
+    // JDK-8232811: to avoid conflicting events, update the geometry only after window pops.
+    if (map_received) {
+        geometry.final_width.value = w;
+        geometry.final_width.type = BOUNDSTYPE_CONTENT;
+        geometry.final_height.value = h;
+        geometry.final_height.type = BOUNDSTYPE_CONTENT;
     }
```

Comment:
So now `geometry.final_width.value` and `geometry.final_height.value` will be set after the `Dialog `is displayed on screen.
Does this mean that, the `Dialog` will be resized after it is displayed ? I did not observe any size change visually.

An observation unrelated to this change: `Dialog` changes it's position after being displayed. It can not be observed always, but is very frequent. Should be easy to observe it on a slow machine.
On the same machine, I did not observe any size change though.

Reply:
No, it will be set on set_bounds(). This will just update de final value after show() / map.